lib/bundler/dsl.rb

```diff
@@ -351,6 +351,10 @@ def normalize_options(name, version, opts)
         opts["source"] = @sources.add_rubygems_source("remotes" => source)
       end
 
+      if opts.key?("force_version") && (r = Gem::Requirement.new(version)) && !r.exact?
+        raise GemfileError, "Cannot use force_version for inexact version requirement `#{r}`"
```

Comment:
I think the error message would be a lot more helpful if it said "force_version can only be used with exact version numbers", does that seem reasonable?